0000-route-broadcasting-protocol/0000-route-broadcasting-protocol.md

```diff
@@ -0,0 +1,226 @@
+
+### Epoch
+Every time the routing table is modified, the update is logged and the revision number of the table is increased. The revision number of the table is called an **epoch**.
+
+A node maintains the epoch that is already known to the counter node, and considers it when requesting route updates so that the node can send only the difference (a stack of adding or withdrawing routes) from the last update with the last known epoch as the starting point. So the node doesn't need to send the whole routing table.
+
+Because the logs of difference could be a large number in some cases, the node may send updates separately multiple times. In that case, "from" and "to" epoch will be specified to express the range of the logs.
+
+![Epoch Logs and Its Update Request](./images/epoch.svg)
+
+The left image shows how epochs are stored. The right image shows an example of a route update request, how "from" and "to" epoch index are given, and how epochs are extracted from the logs.
+
+If the routing table ID that an epoch is based on is changed, the epoch MUST be reset to `0`, so that the counter node can track all the updates from the node.
+
+### Link Relations and Routing
+There are two types of link relations:
+- `Peer` to `Peer`
+- `Parent` to `Child`
+
+In a `Parent` to `Child` relation, the ILP address of the child node is configured automatically using [IL-DCP](../0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md), and generally it is a sub-address of the parent node though it is not a requirement. For instance:
+- `Parent` node
+  - `g.parent-node`
+- `Child` node
+  - `g.parent-node.child-node`
+  - this is a sub-address of `g.parent-node`
+
+In this case, the parent node already knows the address of the child node, so the routing is a truism. The parent node knows when to transfer ILP packets to the child node, and the child node transfers ILP packets basically to the parent node because it is the gateway to the other addresses for the child node.
+
+On the other hand, in a `Peer` to `Peer` relation, the two nodes have different addresses respectively, and the further connected addresses are unclear. So in this case, it is generally helpful to configure to exchange routing information each other.
+
+Conclusively, the following is the general configuration of nodes depending on the relations.
+- `Peer` to `Peer`
+  - Configured to exchange routing information.
+  - Does send route control requests and route update requests that are explained later in the [Requests](#requests) section.
+- `Parent` to `Child`
+  - Configured **NOT** to exchange routing information.
+  - Does **NOT** send route control requests and route update requests.
+  - That said, the nodes MAY be configured to exchange routing information for some specific situations.
+
+### Node Finite State Machine (FSM)
+A node is kind of an FSM that has a status of either:
```

Comment:
Should be clearer that the IDLE/SYNC status is **per counterpart**.

Reply:
Would avoid the wording "kind of" which is imprecise. Would say something like: "Each node consists of a state machine which tracks for each routing counterpart node:"

Reply:
Indeed, my bad. Thank you for the input.